devito/nodes.py

```diff
@@ -297,6 +297,10 @@ def ccode(self):
         """
         loop_body = [s.ccode for s in self.nodes]
 
+        # Disregard offsets for buffered dimensions
+        if self.dim.is_Buffered:
+            self.offsets = [0, 0]
```

Comment:
I agree we shouldn't set this here. Would explicitly ignoring it with something like `offsets = self.offsets if not self.dim.is_Buffered else [0, 0]` and then use `offsets` instead of `self.offsets` work?

Regarding generality, for now we always ignore offsets on `BufferedDimensions`, but not doing this might hold a key to fixing the `pad_time` problem and relates to the overall time indexing issue - so it's likely to change in the future.